hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -938,6 +938,8 @@ private void finishActiveMasterInitialization(MonitoredTask status)
     // Set master as 'initialized'.
     setInitialized(true);
 
+    this.assignmentManager.deleteNonZkBasedQualifiersForZkBasedAssignment();
```

Comment:
I'm not intimately familiar with the startup process, but is it going to be concerning to do this after `setInitialized(true);`? Will there be race conditions where we can start processing assignment before we have cleaned meta?

Reply:
By this time, rebuildUserRegions() would have already built region states in memory (but we have guarding in place). Hence, assignment would not interfere with this operation because `info:sn` and `info:state` won't even be read for ZK based assignment.

Reply:
Sure, but it can be moved before, right?

Reply:
It can be moved before but I am bit inclined to let master finish it's proper initialization before it performs any mutations to meta.